### PR TITLE
Feature/jwt auth

### PR DESCRIPTION
**Pull request by PorterK**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -26,6 +26,7 @@ services:
       - APP_DB_USER=postgres
       - APP_DB_PORT=5432
       - APP_HOST=http://localhost:3000
+      - JWT_SECRET=${JWT_SECRET}
```
Comment (NullVoxPopuli) [inline docker-compose.yml]: no AUTH0?

File: frontend/js/api/AuthenticatedRequest.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        type: method,
+        url: `/api/${endpoint}`,
+        async: false,
+        processData: false,
+        contentType: 'application/json',
+        headers: {
+          'Authorization':`JWT ${store.get('authentication').token}`
+        },
+        data: JSON.stringify(body),
+        success: data => {
```
Comment (NullVoxPopuli) [inline frontend/js/api/AuthenticatedRequest.js]: could be `success: resolve` ?

Comment (PorterK) [inline frontend/js/api/AuthenticatedRequest.js]: Yes :+1: thanks for catching that

File: frontend/js/api/AuthenticatedRequest.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        error: res => {
+          const resp = { status: res.status, error: res.responseText };
+          reject(resp);
+        }
+      })
+    });
+
+  }
+
+  list(){
```
Comment (NullVoxPopuli) [inline frontend/js/api/AuthenticatedRequest.js]: have you seen robovise fetch/resource files? :-)

File: frontend/js/components/Login/Login.js [hunk 4]
```diff
@@ -5,13 +5,13 @@ import { connect } from 'react-redux';
 
 class Login extends Component {
   componentWillMount(){
-    const { getSession } = this.props;
-    getSession();
+    const { validateToken } = this.props;
+    validateToken();
```
Comment (NullVoxPopuli) [inline frontend/js/components/Login/Login.js]: :+1:

**Pull request merged**